src/common/NetworkManager.ts

```diff
@@ -130,14 +161,21 @@ export class NetworkManager extends EventEmitter {
   }
 
   async setOfflineMode(value: boolean): Promise<void> {
-    if (this._offline === value) return;
-    this._offline = value;
+    await this.emulateNetworkConditions(
+      value ? networkConditionsMap['Offline'] : networkConditionsMap['Online']
+    );
+  }
+
+  async emulateNetworkConditions(
+    networkConditions: NetworkConditions
+  ): Promise<void> {
+    if (this._emulatedNetworkConditions === networkConditions) return;
```

Comment:
Note: this can never be `===` when people call `page.emulateNetworkConditions({})` with an object literal (even when it contains exactly the same values). But perhaps that's ok? We expect the most common usage to be 

```js
const slow3G = …;
await page.emulateNetworkConditions(slow3G);
```